public/app/features/alerting/unified/ExistingRuleEditor.tsx

```diff
@@ -19,48 +16,37 @@ interface ExistingRuleEditorProps {
 }
 
 export function ExistingRuleEditor({ identifier, id }: ExistingRuleEditorProps) {
-  useCleanup((state) => (state.unifiedAlerting.ruleForm.existingRule = initialAsyncRequestState));
-
   const {
     loading: loadingAlertRule,
-    result,
+    result: ruleWithLocation,
     error,
-    dispatched,
-  } = useUnifiedAlertingSelector((state) => state.ruleForm.existingRule);
+  } = useRuleWithLocation({ ruleIdentifier: identifier });
 
-  const dispatch = useDispatch();
-  const { isEditable, loading: loadingEditable } = useIsRuleEditable(
-    ruleId.ruleIdentifierToRuleSourceName(identifier),
-    result?.rule
-  );
+  const ruleSourceName = ruleId.ruleIdentifierToRuleSourceName(identifier);
 
-  const loading = loadingAlertRule || loadingEditable;
+  const { isEditable, loading: loadingEditable } = useIsRuleEditable(ruleSourceName, ruleWithLocation?.rule);
 
-  useEffect(() => {
-    if (!dispatched) {
-      dispatch(fetchEditableRuleAction(identifier));
```

Comment:
As per Slack conversation - `fetchEditableRuleAction` is still called once after saving a GMA and remaining on the page (rather than exiting).

We should look into whether we can remove this?